codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -479,4 +274,153 @@ public HttpHeaders clear() {
         super.clear();
         return this;
     }
+
+    @Override
+    public int hashCode() {
+        return size();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof HttpHeaders)) {
+            return false;
+        }
+        HttpHeaders headers = (HttpHeaders) other;
+        if (size() != headers.size()) {
+            return false;
+        }
+        Comparator<CharSequence> valueComparator = AsciiString.CHARSEQUENCE_CASE_SENSITIVE_ORDER;
```

Comment:
@buchgr - I'm fine with having it in `DefaultHeaders`. We can always move it later if it necessary. Thanks!